### DISCUSSION TITLE
Validation fails on Select Relationship

### DISCUSSION THREAD

**Discussion by NanoNulla** [filamentphp/filament#7621] [Help] [answered]
I have the created the models with required relationship but the validation fails when try to submit the form. 

```php
public function form(Form $form): Form
{
    return $form->schema([
            Select::make('customer_id')
                ->relationship('customer', 'full_name')
                ->searchable()
                ->required()
        ])
        ->statePath('data')
        ->model(Purchase::class);
}

public function submit(): void
{
    $this->form->getState()

    dd("here");
}
 ```

FYI, the UI renders correctly but the form validation fails.

**Comment by danharrin** [comment]
No way for us to even start understanding what the issue is, if you don't even tell us what error you are getting

**Reply by NanoNulla** [reply]
@danharrin Thanks for reply, 

When I submit the form, while the customer is already selected but the validation error happens. Here is the screenshot.
![image](https://github.com/filamentphp/filament/assets/21290137/fe31eb1f-1485-430a-bed3-a552a8ff877b)

**Reply by danharrin** [reply, marked as answer]
Do you have `public $data = []` defined? Are you calling `$this->form->fill()` in `mount()`

**Reply by NanoNulla** [reply]
@danharrin 

Yes I have the data property, it was missing the `$this->form->fill()` in `mount()`, fixed the issue. Thanks
